Sources/Clickstream/Dependency/Clickstream/Event/EventChecker.swift

```diff
@@ -0,0 +1,279 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+class EventChecker {
+    static var itemKeySet: Set<String> = []
+
+    /// Check the event name whether valide
+    /// - Parameter eventType: the event name
+    /// - Returns: the EventError object
+    static func checkEventType(eventType: String) -> EventError {
+        if eventType.utf8.count > Event.Limit.MAX_EVENT_TYPE_LENGTH {
+            let errorMsg = """
+            Event name is too long, the max event type length is \
+            \(Event.Limit.MAX_EVENT_TYPE_LENGTH) characters. event name: \(eventType)
+            """
+            log.error(errorMsg)
+            return EventError(errorCode: Event.ErrorCode.EVENT_NAME_LENGTH_EXCEED,
+                              errorMessage: getErrorMessage(errorMsg))
+        } else if !isValidName(name: eventType) {
+            let errorMsg = """
+            Event name can only contains uppercase and lowercase letters, underscores, number,\
+             and is not start with a number. event name: \(eventType)
+            """
+            log.error(errorMsg)
+            return EventError(errorCode: Event.ErrorCode.EVENT_NAME_INVALID,
+                              errorMessage: getErrorMessage(errorMsg))
+        }
+        return EventError(errorCode: Event.ErrorCode.NO_ERROR, errorMessage: "")
+    }
+
+    /// Check the attribute error.
+    /// - Parameters:
+    ///   - currentNumber: current attribute number
+    ///   - key: attribute key
+    ///   - value: attribute value
+    /// - Returns: the ErrorType
+    static func checkAttribute(currentNumber: Int, key: String, value: AttributeValue) -> EventError {
+        if currentNumber >= Event.Limit.MAX_NUM_OF_ATTRIBUTES {
+            let errorMsg = """
+            reached the max number of attributes limit (\(Event.Limit.MAX_NUM_OF_ATTRIBUTES)).\
+             and the attribute: \(key) will not be recorded
+            """
+            log.error(errorMsg)
+            let errorString = "attribute name: \(key)"
+            return EventError(errorCode: Event.ErrorCode.ATTRIBUTE_SIZE_EXCEED,
+                              errorMessage: getErrorMessage(errorString))
+        }
+        let nameLength = key.utf8.count
+        if nameLength > Event.Limit.MAX_LENGTH_OF_NAME {
```

Comment:
extract the EventError out of if clause

only assign the errorMsg and errrorString in if clause

Reply:
fixed